### ISSUE TITLE
Adding support for JavaScript as part of the mainstream languages of PatternHouse

### ISSUE THREAD

**Issue by aryashah2k** [PatternHouse/JavaScript-PatternHouse#1]
Upgrade our suite of Patterns Database by adding code for existing patterns in JavaScript Language.

**Comment by kdgyimah** [comment]
@aryashah2k  Acknowledged.
Could you help us out with a reference guide of some sorts to see the patterns (pseudocode or algorithm) so we can begin?

**Comment by aryashah2k** [comment]
Sure, you can check out the guidelines repo for all the existing patterns for which code is needed
You can add code files with .js extension in the newly created repo
You can also create guides using html (template already provided in the guidelines repo)
Feel free to join the gitter channel where we discuss freely, also hewre is the link to the guidelines repo
https://github.com/PatternHouse/guidelines

**Comment by kdgyimah** [comment]
@aryashah2k  Thanks.
I checked earlier. The link to browse the patterns leads to a 404, See: [https://patternhouse.tk/browse.html](https://patternhouse.tk/browse.html).


Really glad to be part of this too @aryashah2k  :), and together we can build something great!

**Comment by aryashah2k** [comment]
@kdgyimah Yes, the frontend for our website is in progress at the moment, for now you can add codes for the patterns already present in the guidelines repository I have created separate folders for each pattern category and you can start with code contributions and interact with our fellow team member @ChlorinePentoxide who is handling the website part and discuss how you could help with pattern guide pages etc. If there's any questions feel free to ask, glad to have you here!

**Comment by kdgyimah** [comment]
oh, great!

Alright, will check that out.
Anyways, I can help to develop the front end of the site. Is there a design for it?
Also, could you send me a link to it if there is? Thanks

**Comment by aryashah2k** [comment]
Sure, so everything is present right here in the organization's repositories section. Also along with frontend, I highly encourage you to contribute more towards code for various patterns too since very few of the present org members would be there with JavaScript section of the Code contributions. Anyways do join the gitter channel : https://gitter.im/Printing-Pattern-Programs/WebAppDiscussion(I've invited you as well!)
and you can discuss with @ChlorinePentoxide regarding how you can help with the website designing.

For quick reference:
![image](https://user-images.githubusercontent.com/73865728/116900857-dd021c00-ac56-11eb-8bd2-4c1ea9ae0c7a.png)

- https://github.com/PatternHouse/website
- https://github.com/PatternHouse/guides

**Comment by kdgyimah** [comment]
Got it!

> Sure buddy @kdgyimah , so everything is present right here in the organization's repositories section. Also along with frontend, I highly encourage you to contribute more towards code for various patterns too since very few of the present org members would be there with JavaScript section of the Code contributions. Anyways do join the gitter channel : https://gitter.im/Printing-Pattern-Programs/WebAppDiscussion (I've invited you as well!)
> and you can discuss with @ChlorinePentoxide regarding how you can help with the website designing.
> 
> For quick reference:
> ![image](https://user-images.githubusercontent.com/73865728/116900857-dd021c00-ac56-11eb-8bd2-4c1ea9ae0c7a.png)
> 
> * https://github.com/PatternHouse/website
> * https://github.com/PatternHouse/guides
> 
> Link To Website: https://patternhouse.tk

**Comment by ChlorinePentoxide** [comment]
> @aryashah2k Thanks.
> I checked earlier. The link to browse the patterns leads to a 404, See: https://patternhouse.tk/browse.html.
> 
> Really glad to be part of this too @aryashah2k :), and together we can build something great!

The browse page is currently being re-written in Jekyll so we reduce the repetitiveness of the task and focus more on development. It will be up very soon.
